### ISSUE TITLE
Astra python interface crashes with boost exception

### ISSUE THREAD

**Issue by buzmakov** [astra-toolbox/astra-toolbox#9]
I try to process my sinogram using ASTRA.
If I reconstruct only 1 sinogram all works fine.
But if I try reconstruct some slices in the loop (see code below) astra crashes with error message
`terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::bad_lexical_cast> >'  what():  bad lexical cast: source type value could not be interpreted as target `
Sometimes it crashes and sometime all work fine with the same script. This is an example of several simultaneous runs of one script.
```
➜  python astra_tomo2D_test.py
0.00166797637939
0.00754308700562
0.25839304924
➜  python astra_tomo2D_test.py
0.00157809257507
0.00757193565369
terminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::bad_lexical_cast> >'
  what():  bad lexical cast: source type value could not be interpreted as target
➜ python astra_tomo2D_test.py
0.001620054245
0.00748801231384
0.258258104324
```

This as my python script
```python
# coding: utf-8

import astra
import numpy as np
import time


def tomo2d(sinogram, angles):
    detector_size = sinogram.shape[1]

    rec_size = detector_size
    vol_geom = astra.create_vol_geom(rec_size, rec_size)
    proj_geom = astra.create_proj_geom('parallel', 1.0, detector_size, angles)

    sinogram_id = astra.data2d.create('-sino', proj_geom, data=sinogram)
    # Create a data object for the reconstruction
    rec_id = astra.data2d.create('-vol', vol_geom)

    # Set up the parameters for a reconstruction algorithm using the GPU
    cfg = astra.astra_dict('SART_CUDA')
    cfg['ReconstructionDataId'] = rec_id
    cfg['ProjectionDataId'] = sinogram_id
    cfg['option'] = {}
    cfg['option']['MinConstraint'] = 0
    # cfg['option']['MaxConstraint'] = 5

    # Available algorithms:
    # SIRT_CUDA, SART_CUDA, EM_CUDA, FBP_CUDA (see the FBP sample)

    # Create the algorithm object from the configuration structure
    alg_id = astra.algorithm.create(cfg)

    # Run 150 iterations of the algorithm
    astra.algorithm.run(alg_id, 150)

    # Get the result
    rec = astra.data2d.get(rec_id)

    # Clean up. Note that GPU memory is tied up in the algorithm object,
    # and main RAM in the data objects.
    astra.algorithm.delete(alg_id)
    astra.data2d.delete(rec_id)
    astra.data2d.delete(sinogram_id)

    return rec

if __name__ == '__main__':
    t = time.time()
    data = np.load('tomo2d.npz')
    print time.time() - t
    sinogram2 = data['sinogram']
    angles = data['angles']

    detector_size = sinogram2.shape[0]
    res = np.zeros(shape=(detector_size, detector_size), dtype='float32')
    print time.time() - t
    for i in range(10): # works fine for only 1 iteration !!!!
        res = tomo2d(sinogram2, angles)

    print time.time() - t
```
Data file used for this example can be found here 

https://dl.dropboxusercontent.com/u/1592306/tomo2d.npz

I'm using ASTRA from git. Linux Ubuntu 12.04 x64 and 14.04 x64, python 2.7.9
Tested on GPUs NVIDIA GeForce GTX 660M and GTX 580 (CUDA versions 5.5 and 6.5)

Could you help me to fix this issue?
Thank you in advance.

**Comment by wjp** [comment]
Thanks. Looks like one of our cython utility functions for initializing geometries is using data after dropping a reference, and if the garbage collector gets triggered in this (small) interval, it will break. We should have a fix for this soon.

**Comment by wjp** [comment]
As a workaround, you can do

    angles = data['angles'].astype(np.float64)

This will bypass the problematic code.

**Comment by buzmakov** [comment]
Thank you, this workaround work for me.

**Comment by dmpelt** [comment]
Fixed by 26198f7e1fbed9d294fc2856f796f408c85b66f3.
